src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -232,6 +232,22 @@ abstract class UnCurry extends InfoTransform
         // method definition with the same arguments, return type, and body as the original lambda
         val liftedMethod = gen.mkLiftedFunctionBodyMethod(localTyper)(fun.symbol.owner, fun)
 
+        fun match {
+          case Function(self :: params, treeInfo.Applied(core : Ident, _, argss))
+            if core.symbol == self.symbol && argss.flatten.corresponds(params) {
+              case (i : Ident, param) => i.symbol == param.symbol
+              case _ => false
+            } =>
+            reporter.info(fun.pos, "could be a method reference", false)
+          case Function(self :: params, treeInfo.Applied(core : Ident, _, argss))
+            if core.symbol == self.symbol && argss.flatten.corresponds(params) {
+              case (i : Ident, param) => i.symbol == param.symbol
+              case _ => false
+            } =>
+            reporter.info(fun.pos, "could be a method reference", false)
```

Comment:
Is this one meant to be different?

Reply:
I hadn't pushed my latest version. Look again :)